src/main.py

```diff
@@ -42,7 +39,14 @@ def concretize():
                          Static_Con,
                          Has_To_Left_Con, Has_To_Right_Con, Has_Behind_Con, Has_In_Front_Con, Is_Close_To_Con, Is_Medium_Distance_From_Con, Is_Far_From_Con,
                          On_Region_Con,
-                         Behavior_Con, Does_Maneuver_Con, Danger_Con, Collision_Con])
+                         Behavior_Con, Does_Maneuver_Con, Danger_Con, Collision_Con]) 
+    
+    for param in spec.params:
+        args.__dict__[param.key] = param.value
+
+    # 1.1 parse the map file
+    map_file = args.map
```

Comment:
I have to move this line from above to here, because `args` could be modified by the `spec` that was parsed in step 2.0. And things in `spec` will overwrite if there is the same key in the command line argument—is this way okay?

Reply:
Yes, this is fine. eventually, we will find a more systematic way of doing this, but that is work for later.